diff --git a/node_modules/react-native-gesture-handler/android/src/main/jni/CMakeLists.txt b/node_modules/react-native-gesture-handler/android/src/main/jni/CMakeLists.txt
index 026e7ca..9b18799 100644
--- a/node_modules/react-native-gesture-handler/android/src/main/jni/CMakeLists.txt
+++ b/node_modules/react-native-gesture-handler/android/src/main/jni/CMakeLists.txt
@@ -1,3 +1,4 @@
+set(CMAKE_ANDROID_STL_TYPE c++_shared)
 project(GestureHandler)
 cmake_minimum_required(VERSION 3.9.0)
 
@@ -33,8 +34,11 @@ target_include_directories(
 find_package(ReactAndroid REQUIRED CONFIG)
 find_package(fbjni REQUIRED CONFIG)
 
+find_library(CPP_SHARED_LIB c++_shared REQUIRED)
+
 target_link_libraries(
   ${PACKAGE_NAME}
+  ${CPP_SHARED_LIB}
   ReactAndroid::reactnative
   ReactAndroid::jsi
   fbjni::fbjni
