diff --git a/node_modules/react-native-worklets/android/CMakeLists.txt b/node_modules/react-native-worklets/android/CMakeLists.txt
index 9f803d3..328ce8c 100644
--- a/node_modules/react-native-worklets/android/CMakeLists.txt
+++ b/node_modules/react-native-worklets/android/CMakeLists.txt
@@ -4,6 +4,7 @@ cmake_minimum_required(VERSION 3.16)
 set(CMAKE_EXPORT_COMPILE_COMMANDS ON)
 
 set(CMAKE_CXX_STANDARD 20)
+set(CMAKE_ANDROID_STL_TYPE c++_shared)
 
 # Consume shared libraries and headers from prefabs
 find_package(fbjni REQUIRED CONFIG)
@@ -85,6 +86,8 @@ set_target_properties(worklets PROPERTIES LINKER_LANGUAGE CXX)
 
 target_link_libraries(worklets android log ReactAndroid::reactnative
                       ReactAndroid::jsi fbjni::fbjni)
+find_library(CPP_SHARED_LIB c++_shared REQUIRED)
+target_link_libraries(worklets ${CPP_SHARED_LIB})
 
 if(ReactAndroid_VERSION_MINOR GREATER_EQUAL 82)
   target_link_libraries(worklets hermes-engine::hermesvm)
diff --git a/node_modules/react-native-worklets/android/build.gradle.kts b/node_modules/react-native-worklets/android/build.gradle.kts
index 6b80964..a4f354e 100644
--- a/node_modules/react-native-worklets/android/build.gradle.kts
+++ b/node_modules/react-native-worklets/android/build.gradle.kts
@@ -189,6 +189,7 @@ android {
             cmake {
                 arguments(
                     "-DANDROID_STL=c++_shared",
+                    "-DCMAKE_ANDROID_STL_TYPE=c++_shared",
                     "-DANDROID_TOOLCHAIN=clang",
                     "-DREACT_NATIVE_DIR=${toPlatformFileString(reactNativeRootDir.path)}",
                     "-DIS_REANIMATED_EXAMPLE_APP=$IS_REANIMATED_EXAMPLE_APP",
